File: Examples/Ruby/Document/create_PDF_from_XML.rb
require 'aspose_pdf_cloud'

class Document

  include AsposePDFCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @pdf_api = PdfApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def create_PDF_from_XML
    file_name = "newPDFFromXML.pdf"
    template_file = "sample.xsl"
    data_file = "sample.xml"
    upload_file(template_file)
    upload_file(data_file)

    response = @pdf_api.put_create_document(file_name, {template_file: template_file, data_file: data_file, template_type: "xml"})
  end

end

document = Document.new()
puts document.create_PDF_from_XML

File: Examples/PHP/Document/CreatePDFFromXML.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\PDF\PdfApi;
use Aspose\PDF\AsposeApp;

class Document {

	public $pdfApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->pdfApi = new PdfApi();
	}

	public function createPDFFromXML() {
		// Upload file to Aspose Cloud Storage
		$fileName = "newPDFFromXML.pdf";
		$templateFile = "sample.xsl";
		$dataFile = "sample.xml";

		Utils::uploadFile($templateFile);
		Utils::uploadFile($dataFile);

		$result = $this->pdfApi->PutCreateDocument($fileName, $templateFile, $dataFile, $templateType = "xml", $storage = "", $folder = "");
		print_r ( $result );
	}
}

$document = new Document();
$document->createPDFFromXML();

?>
//ExEnd:
